akka-stream/src/main/boilerplate/akka/stream/javadsl/GraphCreate.scala.template

```diff
@@ -20,25 +22,28 @@ private[stream] abstract class GraphCreate {
    * Creates a new [[Graph]] by importing the given graph `g1` and its [[Shape]]
    * along with the [[GraphDSL.Builder]] to the given create function.
    */
-  def create[S1 <: Shape, S <: Shape, M](g1: Graph[S1, M],
+  // TODO DOTTY -> strange...
+  def create[S1 <: Shape @uncheckedVariance, S <: Shape, M](g1: Graph[S1, M],
       block: function.Function2[GraphDSL.Builder[M], S1, S]): Graph[S, M] =
-    scaladsl.GraphDSL.create(g1) { b => s => block.apply(b.asJava, s) }
+    scaladsl.GraphDSL.create(g1) { b => (s: Shape @uncheckedVariance) => block.apply(b.asJava, s.asInstanceOf[S1]) }
```

Comment:
this solves a bunch of errors but is probably wrong

Reply:
In general, when Scala 3 requires `@uncheckedVariance` where Scala 2 doesn't, it's entirely plausible that the Scala 2 compiler was wrongly accepting unsound code. (But that's in general; I haven't dug into this particular case.)

Reply:
I think the original sin here is the definition of Shape with `@uncheckedVariance`:
https://github.com/akka/akka/blob/e8e62e2ca2a0747a2e8fa6b54d205badae988058/akka-stream/src/main/scala/akka/stream/Graph.scala#L23

This breaks soundness as demonstrated by https://scastie.scala-lang.org/smarter/Cno5sLZyTSybDC80uDt2hQ, meaning that we can't really guarantee anything about type-safety in akka.